src/TestingTool/tests/DivideTest.java

```diff
@@ -12,47 +12,122 @@ public DivideTest(UI uiCalculator) {
         this.testingTool = new TestingTool(this.uiCalculator);
     }
 
-    //  Not sure what you were trying to test here below is the function you had
-    /*
-        public int testDivide(int numTests, RelationTransform inRelation, RelationTransform outputRelation) {
-            int numPassed = 0;
-            int a, b;
-            Random rand = new Random();
-            for(int i = 0; i < numTests; i++) {
-                a = rand.nextInt();
-                b = rand.nextInt();
-
-                //Do a / b
-                this.enterCalculatorInput(a);
-                uiCalculator.butDivide.doClick();
-                this.enterCalculatorInput(b);
-                uiCalculator.butEqual.doClick();
-                double output1 = this.getCalculatorOutput();
-                uiCalculator.butCancel.doClick();
-
-                //Metamorphic relation: a/b > 1 if a > b and a/b < 1 if a < b
-                //First Case: a >= b then a/b >= 1
-                //Second Case: a < b, then a/b < 1
-                //Third Case: a > b, b < 0 then a/b < 1
-                //Fourth Case: a < b, b < 0, a < 0 then a/b > 1
-                //Fith Case: a < b, b < 0, a < 0 then a/b >= 1
-                
-                if(a >= b && a/b >= 1 || a < b && a/b < 1 && b > 1 || a > b && a/b < 1 && b < 0 || a > b && b < 0 && a < 0 && a/b < 1 || a < b && b < 0 && a < 0 && a/b >= 1) {
-                    numPassed++;
-                }
-                else
-                {
-                    System.out.println(a + " " + b + " " + (a/b));
-                }
+    //  Test Specific Algebraic Values
+    public boolean[] testAlgebraicValues() {
+        //  Array of [Values to be tested, expected results]
+        double[][] testedValues = {
+            { 10, 1 },
+            { 6, 2 },
+            { 100, 10 },
+            { -6, 3 },
+            { 6.7, 8 },
+            { -2, -20 },
+            { 0.2, -1 },
+            { 12432, 323 },
+            { 5, -3 }
+        };
+
+        double[] testedAnswers = {10,3,10,-2,0.8375,0.1,-0.2,12432.0/323.0, 5.0/-3.0};
```

Comment:
I feel a better way to do this would be to have the answers stored with the tested values in testedValues[i][2]. but should be fine

Reply:
That is a good point, I’ll fix that